lib/streamhidevid-extractor/src/main/java/eu/kanade/tachiyomi/lib/streamhidevidextractor/StreamHideVidExtractor.kt

```diff
@@ -12,25 +12,14 @@ class StreamHideVidExtractor(private val client: OkHttpClient) {
 
     fun videosFromUrl(url: String, prefix: String = ""): List<Video> {
         val page = client.newCall(GET(url)).execute().body.string()
-        val unpacked = JsUnpacker.unpackAndCombine(page) ?: return emptyList()
-        val playlistUrl = unpacked.substringAfter("sources:")
+        val playlistUrl = (JsUnpacker.unpackAndCombine(page) ?: page)
+            .substringAfter("sources:")
             .substringAfter("file:\"") // StreamHide
             .substringAfter("src:\"") // StreamVid
             .substringBefore('"')
-
-        val playlistData = client.newCall(GET(playlistUrl)).execute().body.string()
-
-        val subs = subtitleRegex.findAll(playlistData).map {
-            val urlPart = it.groupValues[2]
-            val subUrl = when {
-                !urlPart.startsWith("https:") ->
-                    playlistUrl.substringBeforeLast("/") + "/$urlPart"
-                else -> urlPart
-            }
-            Track(subUrl, it.groupValues[1])
-        }.toList()
-
-        // The playlist usually only have one video quality.
-        return listOf(Video(playlistUrl, "${prefix}StreamHideVid", playlistUrl, subtitleTracks = subs))
+        if (!playlistUrl.startsWith("http")) return emptyList()
+        return PlaylistUtils(client).extractFromHls(playlistUrl,
```

Comment:
Move playlistutils to a class instance so it doesn't need to be constructed everytime videosFromUrl is called